### ISSUE TITLE
logincredentials not working

### ISSUE THREAD

**Issue by Hans1205** [fdebrus/hayward-ha#1]
Although I can sign in to these websites: 
- https://www.da-gen.com/
- https://vistapool.es/
- https://hayward-europe.web.app/

the credentials do not work in the Aquarite integration. 

Errormessage:
"Unknown error occurred"

What am I doing wrong?

**Comment by keteflips** [comment]
Same problem here

**Comment by fdebrus** [comment]
Works for me, I've just deleted my pool and re-create without issue. When do you get "Unknown error occurred" right after inputing your credentials ? you do not get a screen selection for your pool ? 

so before or after this screen ?

![image](https://github.com/fdebrus/hayward-ha/assets/33791533/8878d84f-9e38-4364-a155-1220b1403e60)

**Comment by fdebrus** [comment]
Try to revert to previous version see if it resolve your problem and keep me posted. the latest has some code that tries to managed the 401 (token expiration) but it's not working yet.

Current way of thinking is that you can't refresh the token as we do not know the clientid and clientsecret, best way forward is to trigger a re-login when token expire not a token refresh.

I need more time to code, given the holiday period is coming in my area, I will be away for 3 weeks.

**Comment by Hans1205** [comment]
> Works for me, I've just deleted my pool and re-create without issue. When do you get "Unknown error occurred" right after inputing your credentials ? you do not get a screen selection for your pool ?
> 
> so before or after this screen ?
> 
> ![image](https://private-user-images.githubusercontent.com/33791533/324258825-8878d84f-9e38-4364-a155-1220b1403e60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3MTc4MDgsIm5iZiI6MTcxMzcxNzUwOCwicGF0aCI6Ii8zMzc5MTUzMy8zMjQyNTg4MjUtODg3OGQ4NGYtOWUzOC00MzY0LWExNTUtMTIyMGIxNDAzZTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIxVDE2MzgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzNDdlMjk5YTRhMWQ3ZjYxYTdkNDA2MzkwNWFlODExYTdmZjk4ZTc3NTgyNDFlMjljODU0YzU5NDY4OTIxNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ozoR5dACwio8pQ4t1k_anZT_KHAQoZpxZU2XJ9QYlTM)

I get this error directly after entering the credentials and before the Pool selection screen

**Comment by Hans1205** [comment]
> Try to revert to previous version see if it resolve your problem and keep me posted. the latest has some code that tries to managed the 401 (token expiration) but it's not working yet.
> 
> Current way of thinking is that you can't refresh the token as we do not know the clientid and clientsecret, best way forward is to trigger a re-login when token expire not a token refresh.
> 
> I need more time to code, given the holiday period is coming in my area, I will be away for 3 weeks.

I tried it just right now with 0.0.5.3 and it works ...

Thank you and have a nice holiday

**Comment by fdebrus** [comment]
Nice, it was the introduction of the code that manages the pool list for accounts having mutiple pools. I just fixed it in 0.0.5.3. 

401 and token refresh is still not fixed, so the integration will stop accepting commands after 1hr (Token expiration time).

If someone wants to look at ? What I will try as time allow is to re-login when a command gets a 401

**Comment by folke** [comment]
The token refresh code doesn't work right now, since `request_url` is not defined [here](https://github.com/fdebrus/hayward-ha/blob/e7fdf40e452cf99187e8d1cb5705dd37bef13a6d/custom_components/aquarite/aquarite.py#L51)

As a result, my logs get spammed with this error, which quickly leads to my HA disk getting full (over 25GB of logging data).

I'll see if I can work on a fix.

Either way, thank you for making this component and enjoy your vacation!

**Comment by fdebrus** [comment]
Thank you, I focussed on the integration and entities for HA. signin and refresh need a full rework, it can be that there is some left over of retry on error, code that make no sense and can be removed.

I've limited knowledge on google API, but I trust we can't refresh token without proper clientid and secret, so re-login shall be the way forward. let's discuss it in the dedicated issue tracker https://github.com/fdebrus/hayward-ha/issues/3

I will close this one, as initial load of pool is resolved for now.
